lib/compat/wordpress-6.8/preload.php

```diff
@@ -13,17 +13,30 @@ function gutenberg_block_editor_preload_paths_6_8( $paths, $context ) {
 	if ( 'core/edit-site' === $context->name || 'core/edit-post' === $context->name ) {
 		$stylesheet       = get_stylesheet();
 		$global_styles_id = WP_Theme_JSON_Resolver_Gutenberg::get_user_global_styles_post_id();
+		$paths[]          = array( '/wp/v2/global-styles/' . $global_styles_id, 'OPTIONS' );
 		$paths[]          = '/wp/v2/global-styles/themes/' . $stylesheet . '?context=view';
 		$paths[]          = '/wp/v2/global-styles/themes/' . $stylesheet . '/variations?context=view';
-		$paths[]          = array( '/wp/v2/global-styles/' . $global_styles_id, 'OPTIONS' );
+
+		// Remove duplicate or unnecessary global styles paths.
 		$excluded_paths[] = '/wp/v2/global-styles/themes/' . $stylesheet;
 		$excluded_paths[] = '/wp/v2/global-styles/' . $global_styles_id;
-	}
-	foreach ( $paths as $key => $path ) {
-		if ( in_array( $path, $excluded_paths, true ) ) {
-			unset( $paths[ $key ] );
+		// Removes the edit context path originating Core's preload paths.
+		$excluded_paths[] = '/wp/v2/global-styles/' . $global_styles_id . '?context=view';
+		// Removes the edit context path originating from gutenberg_block_editor_preload_paths_6_6().
+		$excluded_paths[] = '/wp/v2/global-styles/' . $global_styles_id . '?context=edit';
+		foreach ( $paths as $key => $path ) {
+			if ( in_array( $path, $excluded_paths, true ) ) {
+				unset( $paths[ $key ] );
+			}
+		}
+
+		// Add the global styles path with the correct context based on user caps.
+		if ( 'core/edit-site' === $context->name || 'core/edit-post' === $context->name ) {
+			$context = current_user_can( 'edit_theme_options' ) ? 'edit' : 'view';
+			$paths[] = "/wp/v2/global-styles/$global_styles_id?context=$context";
```

Comment:
Why does this need to run after the excluded paths are removed?

Reply:
The excluded paths are a tiny optimization to remove the preloaded paths we don't need, otherwise the output will be rendered in the HTML. Here we're re-adding only the single one we need based on the user profile.

Reply:
> The excluded paths are a tiny optimization to remove the preloaded paths we don't need, otherwise the output will be rendered in the HTML

Sorry, that's not entirely right. There's no duplication in the rendered HTML, however the `WP_REST_Request` will be fired, even for duplicate paths. So we avoid doing a bunch of expensive `WP_Theme_JSON` unnecessarily.